src/main/java/sg/ncl/MainController.java

```diff
@@ -1554,14 +1587,15 @@ public String adminContributeDataset(Model model) {
     }
     
     @RequestMapping(value="/admin/data/contribute", method=RequestMethod.POST)
-    public String validateAdminContributeDataset(@ModelAttribute("dataset") Dataset dataset, HttpSession session, @RequestParam("file") MultipartFile file, RedirectAttributes redirectAttributes) {
-    	// TODO
+    public String validateAdminContributeDataset(@ModelAttribute("dataset") Dataset dataset, HttpSession session, @RequestParam("file") MultipartFile file, RedirectAttributes redirectAttributes) throws IOException {
+        BufferedOutputStream stream = null;
+        // TODO
     	// validation
     	// get file from user upload to server
 		if (!file.isEmpty()) {
 			try {
 				String fileName = getSessionIdOfLoggedInUser(session) + "-" + file.getOriginalFilename();
-				BufferedOutputStream stream = new BufferedOutputStream(
+				stream = new BufferedOutputStream(
 						new FileOutputStream(new File(App.ROOT + "/" + fileName)));
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Close this "FileOutputStream". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2095)